### PR TITLE
Text carousel and URI extraction

### PR DESCRIPTION
**Pull request by SiAdcock**
picture 8
- Carousel with text headline
- Extracted data URIs out of `embed.js` and into `format` objects
- Made carousel template and event binding more generic, to support both text carousels and big number carousels

### CODE REVIEW

File: src/js/formats/bigNumberCarousel.js [hunk 1]
```diff
@@ -1,129 +1,38 @@
-import bigNumberCarouselTemplate from '../text/bigNumberCarousel.dot.html!text';
-import q from '../lib/query';
+import bigNumberCarouselTemplate from '../text/carousel.dot.html!text';
```
Comment (philwills) [inline src/js/formats/bigNumberCarousel.js]: Should the import name change to reflect that it's not necessarily a big number any more?

**Comment by philwills** [comment]
Not sure I see the value of extracting the URI at the moment, given that it's currently always the same spreadsheet? Is this so that we can inject an alternative URL to test against?

File: src/css/embed.scss [hunk 2]
```diff
@@ -161,12 +161,14 @@ br {
 }
 
 /****** BEGIN Carousel ******/
+$brexit-control-button-size: 28px;
+
 .brexit--carousel {
-    margin: 18px 14px;
+    margin: $baseline*1.5 $brexit-control-button-size*0.5;
```
Comment (philwills) [inline src/css/embed.scss:167]: 👍  for extracting out the `brexit-control-button-size`. Makes things much clearer.

File: src/js/formats/textCarousel.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        const max = 5;
+        let i = 1;
+        let headline;
+        let answer;
+
+        for (; i <= max; i += 1) {
+            headline = data[`headline_${i}`];
+            answer = data[`answer_${i}`];
+            if (
+                (typeof headline !== 'undefined') &&
```
Comment (joelochlann) [inline src/js/formats/textCarousel.js]: any reason why this isn't just `headline && answer`?

File: src/js/formats/helpers/buildCarousel.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import q from '../../lib/query';
+
+export default function buildCarousel(slideCount) {
+    let currentSlide = 0;
+
+    function clearCurrent() {
+        q('.js-current').forEach(function (el) {
```
Comment (joelochlann) [inline src/js/formats/helpers/buildCarousel.js]: could we use ES6 syntax here?

``` javascript
q('.js-current').forEach(el => {
  ...
});
```

**Comment by joelochlann** [comment]
👍 apart from some syntactic pedantry of mine

**Pull request merged**